internal/compiler/parser/document.rs

```diff
@@ -45,6 +46,35 @@ pub fn parse_document(p: &mut impl Parser) -> bool {
                     break;
                 }
             }
+            "@" => {
+                let mut is_export = false;
```

Comment:
So you can add a checkpoint in the parser here and parse the `@rust-attr(...)` node here, and then start the struct node at that checkpoint.
So that the AtRustAttr node would be within the struct